File: Unit14/app/src/main/java/teach/meskills/unit14/Riddles.java
package teach.meskills.unit14;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Random;

public class Riddles extends AppCompatActivity {
    Button showMeRiddle, checkMe;
    TextView someRiddle, finishAnswer;
    EditText personAnswer;
    String choseRiddle = "";
    String[] arrayRiddles = {
            "Жёлтый шар, слегка горчит.\n" +
                    "Летом жажду утолит.",
            "Этот фрукт на вкус хорош\n" +
                    "И на лампочку похож.",
            "Сто одежек -\n" +
                    "Все без застежек",
            "В этот гладкий коробок\n" +
                    "Бронзового цвета\n" +
                    "Спрятан маленький дубок\n" +
                    "Будущего лета."};


    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_riddles);
        showMeRiddle = (Button) findViewById(R.id.show_me_riddles);
        someRiddle = (TextView) findViewById(R.id.some_riddle);
        showMeRiddle.setOnClickListener(v -> {
            shuffleArray(arrayRiddles);
            someRiddle.setText(arrayRiddles[0]);
            choseRiddle = arrayRiddles[0];
        });
        rightAnswerRiddle(choseRiddle);
        personAnswer = (EditText) findViewById(R.id.person_answer);
        finishAnswer = (TextView) findViewById(R.id.finishAnswer);
        checkMe = (Button) findViewById(R.id.checkMe);
        checkMe.setOnClickListener(v -> {
          if(personAnswer.getText().toString().equals(rightAnswerRiddle(choseRiddle))){
              finishAnswer.setText("You are right");
          }
          else{
              finishAnswer.setText("Sorry, it's false, right answer is: " + rightAnswerRiddle(choseRiddle));
          }
        });
    }

    static String rightAnswerRiddle(String choseRiddle){
        String rightAnswer;
        if(choseRiddle.startsWith("Жёлтый")){
            rightAnswer = "грейпфрут";
        }
        else if(choseRiddle.startsWith("Этот фрукт")){
            rightAnswer = "груша";
        }
        else if(choseRiddle.startsWith("Сто")){
            rightAnswer = "капуста";
        }
        else {
            rightAnswer = "желудь";
        }
        return rightAnswer;
    }

    void shuffleArray(String [] arrayRiddles) {
        Random random = new Random();
        for (int i = arrayRiddles.length - 1; i > 0; i--) {
            int index = random.nextInt(i + 1);
            String a = arrayRiddles[index];
            arrayRiddles[index] = arrayRiddles[i];
            arrayRiddles[i] = a;
        }
    }
}

File: Unit14/app/src/main/java/teach/meskills/unit14/Riddles.kt
package teach.meskills.unit14

import androidx.appcompat.app.AppCompatActivity
import android.widget.Button
import android.widget.TextView
import android.widget.EditText
import android.annotation.SuppressLint
import android.os.Bundle
import teach.meskills.unit14.ExampleRiddles.Companion.arrayAnswers
import teach.meskills.unit14.ExampleRiddles.Companion.arrayRiddles

class Riddles : AppCompatActivity() {
    private lateinit var showMeRiddle: Button //кнопка показывает загадку
    private lateinit var checkMe: Button // проверяет правльный ли ответ
    private lateinit var someRiddle: TextView // показывает загадку текст
    private lateinit var rightAnswer: TextView //показывает правильный ли ответ
    private lateinit var personAnswer: EditText //ответ пользователя
    private lateinit var choseRiddle: String //выбранная рандомом загадка
    private var random: Int = 0
    private var right: String = ""

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.riddles_activity)
        showMeRiddle = findViewById(R.id.show_me_riddles)
        someRiddle = findViewById(R.id.some_riddle)
        showMeRiddle.setOnClickListener {
            val shuffle = Shuffle(arrayRiddles().size)
            random = shuffle.doShuffle()
            right = arrayAnswers()[random].toString()
            rightAnswer = findViewById(R.id.rightAnswer)
            choseRiddle = arrayRiddles()[random].toString()
            someRiddle.text = choseRiddle
        }
        checkMe = findViewById(R.id.checkMe)
        checkMe.setOnClickListener {
            personAnswer = findViewById(R.id.person_answer)
            if (personAnswer.text.toString() == right) {
                rightAnswer.text ="You are right, answer is $right"
            } else {
                rightAnswer.text ="Sorry, right answer is $right"
            }
        }
    }
}

